models/document.py

```diff
@@ -0,0 +1,151 @@
+from nltk.tokenize import word_tokenize
+
+
+class DocumentCollection(object):
+
+    def __init__(self, data_filename='', stop_list_filename='', load_on_creation=True):
+        self.data_filename = data_filename
+        self.stop_list_filename = stop_list_filename
+        self.common_words = []
+        self.collection = {}
+        if load_on_creation:
+            self.load_common_words()
+            self.load_collection()
+
+    def load_common_words(self):
+        with open(self.stop_list_filename, 'r') as sl:
+            data = sl.read()
+
+        self.common_words = [line.strip() for line in data.split('\n')]
+
+    def load_collection(self):
+        raise NotImplementedError
+
+    @property
+    def vocabulary(self):
+        vocabulary = {}
+        for document in self.collection.values():
+            for attr in document.tokenized_fields:
+                for token in getattr(document, attr):
+                    try:
+                        vocabulary[token] += 1
+                    except KeyError:
+                        vocabulary[token] = 1
+
+        return vocabulary
+
+    @property
+    def vocabulary_size(self):
+        return len(self.vocabulary.keys())
+
```

Comment:
Is this function efficient ? Each time we want to know the size of the vocabulary, we have to go through all the collection.
I think at the end of vocabulary function, we should store the vocabulary size.

Reply:
Agree. Behaviour changed.